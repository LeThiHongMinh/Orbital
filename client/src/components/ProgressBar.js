import React, { useState } from 'react';
import { LinearProgress, Typography, TextField } from '@mui/material';
import { deve } from '../assets/images';

const ProgressBar = () => {
  const [progress, setProgress] = useState(0); // Initial progress value
  const [inputValue, setInputValue] = useState(''); // State to hold the input value

  // Function to handle input change
  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  // Function to handle form submission
  const handleSubmit = (e) => {
    e.preventDefault();
    setProgress(Number(inputValue)); // Convert input value to number and set progress
    setInputValue(''); // Clear input field after submission
  };

  return (
    <div className="fixed top-0 left-0 right-0 flex items-center justify-center p-6">
      <div className="bg-red-200 p-6 rounded-lg shadow-lg max-w-3xl w-full">
        <img
          src={deve}
          alt="NUSTudy Logo"
          width={200}
          height={200}
          className="mb-4 mx-auto"
        />
        <LinearProgress
          variant="determinate"
          value={progress}
          sx={{
            height: 30, // Increased height
            borderRadius: 8,
            '& .MuiLinearProgress-bar': {
              borderRadius: 8,
            },
          }}
        />
        <Typography variant="h6" color="textSecondary" align="center" className="mt-2">
          Progress: {progress}%
        </Typography>
        <form onSubmit={handleSubmit} className="mt-4">
          <TextField
            label="Enter Progress (%)"
            variant="outlined"
            type="number"
            value={inputValue}
            onChange={handleInputChange}
            fullWidth
          />
        </form>
      </div>
    </div>
  );
};

export default ProgressBar;
