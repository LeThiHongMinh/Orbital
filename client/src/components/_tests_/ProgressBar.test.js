import * as React from 'react'
import ProgressBar from '../ProgressBar'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer.create(<ProgressBar />)
  expect(tree).toMatchSnapshot()
})