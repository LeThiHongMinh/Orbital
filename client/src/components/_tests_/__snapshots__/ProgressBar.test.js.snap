// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="fixed top-0 left-0 right-0 flex items-center justify-center p-6"
>
  <div
    className="bg-red-200 p-6 rounded-lg shadow-lg max-w-3xl w-full"
  >
    <img
      alt="NUSTudy Logo"
      className="mb-4 mx-auto"
      height={200}
      src="deve.png"
      width={200}
    />
    <span
      aria-valuemax={100}
      aria-valuemin={0}
      aria-valuenow={0}
      className="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate css-13vmswx-MuiLinearProgress-root"
      role="progressbar"
    >
      <span
        className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate css-5xe99f-MuiLinearProgress-bar1"
        style={
          Object {
            "transform": "translateX(-100%)",
          }
        }
      />
    </span>
    <h6
      className="MuiTypography-root MuiTypography-h6 MuiTypography-alignCenter mt-2 css-1oi36s0-MuiTypography-root"
    >
      Progress: 
      0
      %
    </h6>
    <form
      className="mt-4"
      onSubmit={[Function]}
    >
      <div
        className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          htmlFor=":r0:"
          id=":r0:-label"
        >
          Enter Progress (%)
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            disabled={false}
            id=":r0:"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="number"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-yjsfm1"
            >
              <span>
                Enter Progress (%)
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </form>
  </div>
</div>
`;
